11.bulk_Morphology_Elastic_Network_LOCO/0.split_data/scripts/0.split_data_regression.py

```diff
@@ -0,0 +1,232 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# In[1]:
+
+
+import argparse
+import itertools
+import pathlib
+
+import numpy as np
+import pandas as pd
+import pyarrow.parquet as pq
+import toml
+from joblib import dump
+from sklearn.exceptions import ConvergenceWarning
+from sklearn.linear_model import LogisticRegression
+from sklearn.model_selection import GridSearchCV, StratifiedKFold, train_test_split
+
+# In[2]:
+
+
+argparser = argparse.ArgumentParser()
+argparser.add_argument("--cell_type", default="all")
+
+args = argparser.parse_args()
+
+cell_type = args.cell_type
+
+
+# In[3]:
+
+
+# Parameters
+aggregation = True
+nomic = True
+
+
+# In[4]:
+
+
+MODEL_TYPE = "regression"
+
+
+# In[5]:
+
+
+# toml file path
+TOML_PATH = pathlib.Path("../splits.toml")
+# read toml file via toml
+data_splits_by_treatments = toml.load(TOML_PATH)
+
+# define the 100% test set data treatments
+test_100_percent = data_splits_by_treatments["splits"]["data_splits_100"]
+test_75_percent = data_splits_by_treatments["splits"]["data_splits_75"]
+
+
+# In[6]:
+
+
+path = pathlib.Path(
+    f"../../../data/{cell_type}_preprocessed_sc_norm_aggregated_nomic.parquet"
+)
+
+data_df = pd.read_parquet(path)
+
+data_df.head()
+
+
+# In[7]:
+
+
+# subset each column that contains metadata
+metadata = data_df.filter(regex="Metadata")
+
+# get all columns that are not metadata except for metadata_Well
+data = data_df.drop(metadata.columns, axis=1)
+
+# get the metadata_Well column
+metadata_well = metadata[
+    ["Metadata_Well", "oneb_Metadata_Treatment_Dose_Inhibitor_Dose"]
+]
+
+data_df = pd.merge(data, metadata_well, left_index=True, right_index=True)
+
+
+# This model and code is both inspired and reused from: https://github.com/WayScience/phenotypic_profiling_model/blob/main/1.split_data/split_data.ipynb
+# The bulk of this work was done by **Roshan Kern** I have only made minor changes to the code to make it more modular and easier to use for my purposes.
+
+# In[8]:
+
+
+# variable test and train set splits
+# 100% test set
+# subset the following treatments for test set
+test_set_all = data_df[
+    data_df["oneb_Metadata_Treatment_Dose_Inhibitor_Dose"].isin(test_100_percent)
+]
+# 75% test set and 25% train set
+test_set_75 = data_df[
+    data_df["oneb_Metadata_Treatment_Dose_Inhibitor_Dose"].isin(test_75_percent)
+]
+
+test_100_and_75 = test_100_percent + test_75_percent
+
+# 50% test set and 50% train set
+# get all treatments that are not in the_test_set_all and the test_set_75
+test_set_50 = data_df[
+    ~data_df["oneb_Metadata_Treatment_Dose_Inhibitor_Dose"].isin(test_100_and_75)
+]
+
+print(test_set_all.shape, test_set_75.shape, test_set_50.shape)
+
+
+# In[9]:
+
+
+# get the train test splits from each group
+# 100% test set
+test_set_all
```

Comment:
will this print out a lot of stuff? maybe not worth doing?

Reply:
Yes, I removed this line